sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -903,9 +907,11 @@ def on_port_update_event(self, port_change_event):
             return
 
         # Skip if the port/cage type is not a CMIS
-        ptype = _wrapper_get_sfp_type(pport)
-        if ptype not in self.CMIS_MODULE_TYPES:
-            return
+        # 'index' can be -1 if STATE_DB|PORT_TABLE
+        # TODO:Make sure doesnot breakt non-CMIS
+        #ptype = _wrapper_get_sfp_type(pport)
```

Comment:
I believe get_sfp_type() returns the sfp_type present in the port and not the port cage type. 
So can you rename the variable and uncomment the code?

Reply:
get_sfp_type() returns only the port/cage type, and it's a constant initialized at chassis.__init__()
e.g.
https://github.com/Azure/sonic-buildimage/blob/master/platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py#L132
And a QSFP28 could be attached to the QSFPDD port on the box, and it's better to skip the CMIS init sequence for it.

Reply:
@ds952811 form-factor of the transceiver is not the right check to determine whether the module follows CMIS or not. Even QSFP form-factor may follow CMIS https://github.com/Azure/sonic-platform-common/pull/246 that follows CMIS